### PR TITLE
UTILS: Add ucs machine id

### PR DESCRIPTION
**Pull request by lappazos**
## What
Add use of UCS ucs_get_system_id in proc_info 

## Why ?
More accurate machine id in various cases

## How ?
Check for function existence in UCS, is yes - use it, else keep same.

### CODE REVIEW

**Comment by lappazos** [comment]
according to PR https://github.com/openucx/ucx/pull/7180

File: src/utils/ucc_proc_info.c [hunk 1]
```diff
@@ -3,6 +3,10 @@
 #include "ucc_log.h"
 #include <limits.h>
 #include <stdint.h>
+#include "config.h"
```
Comment (vspetrov) [inline src/utils/ucc_proc_info.c:6]: i remember we decided to put all the UCS stuff to utils. So, lets create ucc_get_system_id function, and implement it there via ucs_get_system_id.

File: src/utils/ucc_proc_info.c [hunk 2]
```diff
@@ -21,8 +33,11 @@ ucc_status_t ucc_local_proc_info_init()
     } else {
         strtok(ucc_local_hostname, ".");
         ucc_assert(sizeof(ucc_host_id_t) >= sizeof(unsigned long));
-        //TODO: switch to ucs_machine_guid when it is available
-        ucc_local_proc.host_hash = ucc_str_hash_djb2(ucc_local_hostname);
+#ifdef HAVE_UCS_GET_SYSTEM_ID
```
Comment (vspetrov) [inline src/utils/ucc_proc_info.c:36]: don't need ifdef here

**Review by vspetrov — CHANGES_REQUESTED** [review]

File: config/m4/ucx.m4 [hunk 3]
```diff
@@ -56,6 +56,16 @@ AS_IF([test "x$ucx_checked" != "xyes"],[
...
             ucx_happy="no"
         ])
+        
+        AC_CHECK_HEADERS([ucs/sys/uid.h],
+        [
+            AC_CHECK_LIB([ucs], [ucs_get_system_id],
+            [
+                AC_DEFINE([HAVE_UCS_GET_SYSTEM_ID], 1, [Enable use of ucs unique machine identifier])
+            ],
+            [],[-luct -lucm -lucp])
```
Comment (bureddy) [inline config/m4/ucx.m4:66]: is "ucm" "ucp"  needed here?

**Review by bureddy — APPROVED** [review]

**Review by vspetrov — APPROVED** [review]

**Comment by manjugv** [comment]
@lappazos There are some CI errors. Do you have details for it?

**Comment by lappazos** [comment]
Waits for https://github.com/openucx/ucc/pull/287 in order to pass tests

**Comment by lappazos** [comment]
> @lappazos There are some CI errors. Do you have details for it?
Waits for #287 in order to pass tests

File: config/m4/ucx.m4 (hunk 3 above)
Comment (lappazos) [reply config/m4/ucx.m4:66]: it won't hurt, but might prevent errors

**Pull request merged**